### PR TITLE
[gsoc] Add PODIO proposal

### PR DESCRIPTION
**Pull request by jblomer**
Draft proposal for the PODIO/RNTuple connection. Thomas and Valentin should approve before merging.

### CODE REVIEW

**Review by xvallspl — CHANGES_REQUESTED** [review]
Valentin and Thomas are not in mentors.md

**Comment by jblomer** [comment]
Thanks! Added.

**Review by aflp91 — APPROVED** [review]
OK with me!

File: _gsocproposals/2021/proposal_podio.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+While PODIO supports ROOT's current production I/O subsystem ("TTree"), no such support exists yet for RNTuple.
+The goal of this project is to add RNTuple support to PODIO.
+
+## Task ideas and expected results
+The project should develop a podio writer and reader that uses the ROOT RNTuple API in order to serialize and deserialize vectors of PODs.
+The existing TTree reader and writer can be used as a template.
+A first comparison between RNTuple and TTree files (e.g., file size, read and write throughput) would complete the project.
+
+## Evaluation Task
+Interested students please contact Jakob (jblomer@cern.ch) or Javier (j.lopez@cern.ch) for a warm-up and evaluation task.
```
Comment (tmadlener) [inline _gsocproposals/2021/proposal_podio.md:29]:
```suggestion
Interested students please contact Jakob (jblomer@cern.ch), Valentin (valentin.volkl@cern.ch) or Thomas (thomas.madlener@desy.de) for a warm-up and evaluation task.
```

**Review by agheata — APPROVED** [review]

File: _gsocproposals/2021/proposal_podio.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+organization:
+    - DESY
+    - CERN
+---
+
+## Description
+PODIO is a C++ toolkit for the creation of event data models (EDMs) with a fast and efficient I/O layer.
+An event data model is like a schema: it describes the structure of the data collected by a particle detector.
+Event data models typically consist of collections of records that describe physics objects identified by the detector, for instance particles, tracks, jets, and clusters.
+PODIO translates such high-level object hierarchies into vectors of plain-old-data (POD) data structures for optimal I/O performance.
```
Comment (tmadlener) [inline _gsocproposals/2021/proposal_podio.md:15]:
```suggestion
PODIO offers automatic code generation for efficiently implementing the EDM starting from a high level description of the objects.
PODIO uses plain-old-data (POD) structures to store the data of the objects and translates hierarchies of objects into vector of PODs for optimal I/O performance.
```

Comment (tmadlener) [reply _gsocproposals/2021/proposal_podio.md:15]: Just to mention that there is a code generation step somewhere in PODIO.

**Review by tmadlener — CHANGES_REQUESTED** [review]
Email addresses need to be updated to reflect the mentors here.

**Comment by agheata** [comment]
@jblomer thanks for addressing the open discussions. If you need also a review from Valentin you should ping him.

**Review by vvolkl — APPROVED** [review]

**Review by tmadlener — APPROVED** [review]
Thanks for preparing this.

**Pull request merged**